crates/oci/src/client.rs

```diff
@@ -230,10 +300,51 @@ impl Client {
                             .await
                         {
                             Err(e) => return Err(e),
+                            // TODO: handle error(s), avoid unwraps below
                             _ => match layer.media_type.as_str() {
                                 WASM_LAYER_MEDIA_TYPE => {
                                     let _ = this.cache.write_wasm(&bytes, &layer.digest).await;
                                 }
+                                ARCHIVE_MEDIATYPE => {
+                                    // Write archive layer as usual to get cache check
+                                    let _ = this.cache.write_data(&bytes, &layer.digest).await;
```

Comment:
I see you just copied from above, but this should definitely be checking its result (here and above):

```suggestion
                                    this.cache.write_data(&bytes, &layer.digest).await?;
```

Reply:
I'm seeing the following when adding the `?` to bubble up the result:
<img width="1386" alt="Screenshot 2023-10-10 at 3 41 00 PM" src="https://github.com/fermyon/spin/assets/9322017/7e6d4179-0262-4ec6-a637-744365d43625">

Which led me to something like the following, but that doesn't look pretty:
<img width="823" alt="Screenshot 2023-10-10 at 3 41 10 PM" src="https://github.com/fermyon/spin/assets/9322017/dbe6a996-887a-4a6e-8892-6e102f7a6260">

There has to be a better way! (Right?)

Reply:
Slightly nicer... maybe: [64f0ae3](https://github.com/fermyon/spin/pull/1847/commits/64f0ae3b2e92d89f814331a60c5c7208c15da35a)